src/components/form/CreateModelForm.vue

```diff
@@ -99,12 +102,35 @@ const props = defineProps({
 });
 
 const plugins = reactive(getPlugins());
-const modelName = ref(plugins[0].configuration.defaultFileName);
+const modelPath = ref();
 const modelPlugin = ref(plugins[0]?.data.name);
 const submitting = ref(false);
 const models = ref([]);
-const allFolders = ref([]);
-const baseFolder = computed(() => getPluginByName(modelPlugin.value).configuration.restrictiveFolder || '');
+const pluginConfiguration = computed(() => getPluginByName(modelPlugin.value).configuration);
+const fileName = computed(() => pluginConfiguration.value.defaultFileName || '');
+const baseFolder = computed(() => pluginConfiguration.value.restrictiveFolder || '');
+const canCreateRootModel = computed(() => pluginConfiguration.value.restrictiveFolder === null);
+const modelLocation = computed(() => {
+  if (pluginConfiguration.value.isFolderTypeDiagram) {
+    if (modelPath.value?.length > 0) {
+      return `${baseFolder.value}${modelPath.value}/${fileName.value}`;
+    }
+    return `${baseFolder.value}${fileName.value}`;
+  }
+
+  return `${baseFolder.value}${modelPath.value}`;
+});
+
+/**
+ * Check if new diagram to create has a valid path.
+ * @returns {boolean | string} Return true if the value is a valid diagram path,
+ * otherwise the translated error message.
+ */
+function isValidDiagramPath() {
+  return getPluginByName(modelPlugin.value)
+    .isParsable(new FileInformation({ path: modelLocation.value }))
+    ? null : t('errors.models.notParsable');
```

Comment:
should return a bool according to the description.

Reply:
done